### PR TITLE
Documentation for the serve_hostnames soak test

### PR DESCRIPTION
**Pull request by satnam6502**

### CODE REVIEW

**Comment by satnam6502** [comment]
FYI @alex-mohr in case this can be used as soak test for GKE (or adapted in some way -- happy to help).

File: test/soak/serve_hostnames/README.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
+# Soak Test serve_hostnames
+This directory contains the source for a soak test `server_hostnames` which performs the following actions when used with the GCE provider:
```
Comment (zmerlynn) [inline test/soak/serve_hostnames/README.md]: s/server_hostnames/serve_hostnames/

**Comment by zmerlynn** [comment]
LGTM modulo nit

File: test/soak/serve_hostnames/README.md (hunk 1 above)
Comment (satnam6502) [inline test/soak/serve_hostnames/README.md]: Fixed, thanks. Every single serve_hostnames was originally typed as server_hostnames and then fixed up. I missed this one.

**Pull request merged**